applicationset/services/scm_provider/bitbucket_server.go

```diff
@@ -183,8 +183,8 @@ func (b *BitbucketServerProvider) listBranches(repo *Repository) ([]bitbucketv1.
 
 func (b *BitbucketServerProvider) getDefaultBranch(org string, repo string) (*bitbucketv1.Branch, error) {
 	response, err := b.client.DefaultApi.GetDefaultBranch(org, repo)
-	if response != nil && response.StatusCode == 404 {
-		// There's no default branch i.e. empty repo, not an error
+	// The API will return 404 if a default branch is set but doesn't exist or nil in case the default branch is not set (empty repo)
+	if (response != nil && response.StatusCode == 404) || (response == nil) {
```

Comment:
@jjsiv I'm worried that this is going to incorrectly capture instances where there's a proper error (i.e. `response` is nil for some reason _besides_ an empty repo, and `err` contains info about the actual failure).

Can you re-craft the conditions in a way that ensures we're only returning `nil, nil` for the empty repo case?

Reply:
Right, and it does actually return an error (EOF) alongside a nil response. Perhaps we could explicitly check for that - do you think this would be an appropriate solution?
`if (response != nil && response.StatusCode == 404) || (response == nil && err == io.EOF)`

Reply:
Yep, that should do it!